consensus/src/chained_bft/chained_bft_smr.rs

```diff
@@ -159,54 +201,56 @@ impl<T: Payload> ConsensusProvider for ChainedBftSMR<T> {
             .enable_all()
             .build()
             .expect("Failed to create Tokio runtime!");
-
-        let initial_data = runtime.block_on(self.storage.start());
         let input = self.input.take().expect("already started, input is None");
 
         let executor = runtime.handle().clone();
         let time_service = Arc::new(ClockTimeService::new(executor.clone()));
-        self.runtime = Some(runtime);
 
         let (timeout_sender, timeout_receiver) =
             channel::new(1_024, &counters::PENDING_PACEMAKER_TIMEOUTS);
         let (self_sender, self_receiver) = channel::new(1_024, &counters::PENDING_SELF_MESSAGES);
-        let epoch_info = Arc::new(RwLock::new(EpochInfo {
-            epoch: initial_data.epoch(),
-            verifier: initial_data.validators(),
-        }));
+
+        let liveness_storage_data: LivenessStorageData<T> = runtime.block_on(self.storage.start());
+
+        let epoch_info = liveness_storage_data.epoch_info();
 
         let mut epoch_mgr = EpochManager::new(
             self.author,
             Arc::clone(&epoch_info),
             input.config,
             time_service,
             self_sender,
-            input.network_sender,
+            input.network_sender.clone(),
             timeout_sender,
             input.txn_manager,
             input.state_computer,
             self.storage.clone(),
             input.safety_rules_manager,
         );
 
-        // Step 2
-        let event_processor = epoch_mgr.start_epoch(initial_data);
-
-        // TODO: this is test only, we should remove this
-        self.block_store = Some(event_processor.block_store());
-
         let (network_task, network_receiver) =
             NetworkTask::new(epoch_info, input.network_events, self_receiver);
 
+        let processor = match epoch_mgr.start_epoch(liveness_storage_data) {
```

Comment:
```
if let Processor::EventProcessor(p) = processor {
  self.block_store = Some(p.block_store());
}
```

Reply:
I can't take the processor out right here because it needs to be passed to `start_event_processing`

Reply:
we can use reference instead